### PR TITLE
Add patches for gcc5 compile breaks

### LINKED ISSUE #3331: [package] folly/2020.08.10.00: Compile errors on devtoolset-4 gcc5
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **folly/2020.08.10.00**
  * Operating System+version: **Red Hat Enterprise 6**
  * Compiler+version: **GCC 5.3.1 from devtoolset-4**
  * Conan version: **conan 1.29.1**
  * Python version: **Python 3.7.4**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=5
compiler.libcxx=libstdc++
build_type=Debug
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)

Build recipe using gcc 5.3.1 (devtoolset-4) on RHEL6.  There are two compile errors:

```
/userhome/epederson/.conan/data/folly/2020.08.10.00/test/test/build/21e92cd0e529b036c562d4dfa4ad1adfee8a5c75/source_subfolder/folly/portability/Time.cpp:278:2: error: #error No clock_gettime(3) compatibility wrapper available for this platform.
 #error No clock_gettime(3) compatibility wrapper available for this platform.
  ^
gmake[2]: *** [source_subfolder/CMakeFiles/folly_base.dir/folly/portability/Time.cpp.o] Error 1
```

```
/userhome/epederson/.conan/data/folly/2020.08.10.00/test/test/build/21e92cd0e529b036c562d4dfa4ad1adfee8a5c75/source_subfolder/folly/Benchmark.cpp:340:30: error: expected ‘)’ before ‘PRId64’
               printf("  %-*" PRId64, int(name.length()), ptr->value);
                              ^
```

I will submit a PR with a fix.

### PR DESCRIPTION
**Pull request by sourcedelica**
Specify library name and version:  **folly/2020.08.10**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Fixes #3331

### CODE REVIEW

File: recipes/folly/all/patches/0014-find-librt.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+diff --git a/CMake/FollyConfigChecks.cmake b/CMake/FollyConfigChecks.cmake
+index 6b8b308c7..908d72d51 100644
+--- a/CMake/FollyConfigChecks.cmake
++++ b/CMake/FollyConfigChecks.cmake
+@@ -83,6 +83,13 @@ string(REGEX REPLACE
+   CMAKE_REQUIRED_FLAGS
+   "${CMAKE_REQUIRED_FLAGS}")
+ 
++if (UNIX)
```
Comment (madebr) [inline recipes/folly/all/patches/0014-find-librt.patch:9]: https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html#variable:CMAKE_SYSTEM_NAME
```suggestion
+if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
```

**Review by SSE4 — DISMISSED** [review]

File: recipes/folly/all/patches/0014-find-librt.patch (hunk 1 above)
Comment (sourcedelica) [reply recipes/folly/all/patches/0014-find-librt.patch]: Good point - thanks

File: recipes/folly/all/patches/0014-find-librt.patch [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+--- a/CMake/FollyConfigChecks.cmake
++++ b/CMake/FollyConfigChecks.cmake
+@@ -83,6 +83,13 @@ string(REGEX REPLACE
+   CMAKE_REQUIRED_FLAGS
+   "${CMAKE_REQUIRED_FLAGS}")
+ 
++if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
++  find_library(LIBRT rt)
++  if (LIBRT)
++    set(CMAKE_REQUIRED_LIBRARIES "rt")
```
Comment (madebr) [inline recipes/folly/all/patches/0014-find-librt.patch:12]: Should `rt` be appended to the list, or replace it?
```suggestion
+    list(APPEND CMAKE_REQUIRED_LIBRARIES "rt")
```

Comment (sourcedelica) [reply recipes/folly/all/patches/0014-find-librt.patch:12]: `CMAKE_REQUIRED_LIBRARIES` is used for other libraries like zlib and OpenSSL: https://github.com/facebook/folly/search?q=CMAKE_REQUIRED_LIBRARIES.  I didn't want to disturb those.

Comment (madebr) [reply recipes/folly/all/patches/0014-find-librt.patch:12]: By doing a plain `set`, you overwrite the previous content.

Comment (sourcedelica) [reply recipes/folly/all/patches/0014-find-librt.patch]: Doh.  That's what I had intended to to.  Fixed.

Comment (madebr) [reply recipes/folly/all/patches/0014-find-librt.patch]: Should `rt` also be added `system_libs` in folly's conanfile.

Comment (sourcedelica) [reply recipes/folly/all/patches/0014-find-librt.patch]: Yes - done.

**Review by madebr — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**